build-logic/convention/src/main/kotlin/AndroidLibraryConventionPlugin.kt

```diff
@@ -42,6 +42,7 @@ class AndroidLibraryConventionPlugin : Plugin<Project> {
                 defaultConfig.targetSdk = 33
                 configureFlavors(this)
                 configureGradleManagedDevices(this)
+                resourcePrefix = path.split("""\W""".toRegex()).drop(1).distinct().joinToString(separator = "_").lowercase() + "_"
```

Comment:
```suggestion
                // The resource prefix is derived from the module name,
                // so resources inside ":core:module1" must be prefixed with  "core_module1_" 
                resourcePrefix = path.replace(":([\\w\\-]+)".toRegex(), "$1_")
```

Reply:
Really nice idea, thanks. 

I simplified your regex, the only difference being that dashes in the module name are retained, rather than being replaced for underscores.

Now all that needs to be done is to update all the resource names to match, which is a fairly large change. Would you be able to do this? There's a chance this might cause some breaking changes for in-progress PRs, but hopefully not.

Reply:
@dturner I've update the PR with the new resource names. There was one "error" [here](https://github.com/android/nowinandroid/pull/767/commits/6c941ccd96e75eab2b812523197dd6c2c88906dd#diff-9df87753608fc53600f9e148c0e20ee6f0a40520c39283a2f4be7f5cf64ca16fL24-R24).
And I've taken the liberty to rename some resources where it would have produce duplicated words, as well as "title" strings to be more explicit.
Should we keep the current regex or should we find a "simpler" (smaller) regex?

Reply:
Let's keep your version, it just took me a while to figure out what it was doing :)